src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -243,10 +243,15 @@ export class JupyterSession extends BaseJupyterSession {
             await this.kernelService.ensureKernelIsUsable(kernelConnection, cancelToken, disableUI);
         }
 
+        // If kernelName is empty this can cause problems for servers that don't
+        // understand that empty kernel name means the default kernel.
+        // See https://github.com/microsoft/vscode-jupyter/issues/5290
+        const kernelName = getNameOfKernelConnection(kernelConnection) ?? this.sessionManager?.specs?.default ?? '';
```

Comment:
What value do you get (with remote jupyter)?

Reply:
With AML remote compute the kernelConnection isn't undefined but `this.sessionManager?.specs?.default` resolves to the name of the default kernel on the remote (matches up with `this.sessionManager?.specs?.kernelspecs[this.sessionManager?.specs?.default].name`).

Reply:
This is what `this.sessionManager.specs` is for AML 
![image](https://user-images.githubusercontent.com/30305945/116303244-bcb5f700-a756-11eb-8a02-343f06573079.png)

Reply:
Perfect. I think that should work for Synapse. Hopefully @zesluo can try this out once it goes through.